### PR TITLE
adds support for glob-style input spec paths and directory output paths

### PR DESCRIPTION
**Pull request by sharmarajdaksh**
This PR adds support for glob-style inputs and directory output paths for the openapi-typescript cli, as mentioned in #526.
With this, something like the following should be possible:
```
npx openapi-typescript specs/*.yaml -o outdir/
```
- Maintains backwards compatibility, and throws reasonable helpful errors.
- If the input is a glob path and -o flag it specified, it must be a directory
- Cannot specify remote resources using glob-type paths

### CODE REVIEW

File: tests/bin/cli.test.ts [hunk 1]
```diff
@@ -31,4 +31,14 @@ describe("cli", () => {
...
+
+  it("supports glob paths", () => {
+    const expectedPetstore = fs.readFileSync(path.join(__dirname, "expected", "petstore.ts"), "utf8");
+    const expectedManifold = fs.readFileSync(path.join(__dirname, "expected", "manifold.ts"), "utf8");
+    execSync(`../../bin/cli.js \"specs/*.yaml\" -o generated/`, { cwd: __dirname }); // Quotes are necessary because shells like zsh treats glob weirdly
+    const outputPetstore = fs.readFileSync(path.join(__dirname, "generated", "petstore.ts"), "utf8");
+    const outputManifold = fs.readFileSync(path.join(__dirname, "generated", "manifold.ts"), "utf8");
+    expect(outputPetstore).toBe(expectedPetstore);
+    expect(outputManifold).toBe(expectedManifold);
+  });
```
Comment (drwpow) [inline tests/bin/cli.test.ts:43]: Nice test!

File: bin/cli.js [hunk 2]
```diff
@@ -104,15 +143,19 @@ async function main() {
...
-    const timeEnd = process.hrtime(timeStart);
-    const time = timeEnd[0] + Math.round(timeEnd[1] / 1e6);
-    console.log(green(`🚀 ${pathToSpec} -> ${bold(cli.flags.output)} [${time}ms]`));
-  } else {
-    // output option 2: stdout
-    process.stdout.write(result);
+  let result = "";
+  for (const specPath of inputSpecPaths) {
+    // append result returned for each spec
+    result += await generateSchema(specPath);
```
Comment (drwpow) [inline bin/cli.js:158]: This is clever. I hadn’t thought about the `stdout` case but this seems pretty handy.

Comment (drwpow) [reply bin/cli.js:158]: Oh wait—we’re logging to `stdout` elsewhere. Truth be told, I don’t know why we’re returning the `result` here; it doesn’t seem to be helpful (and possibly wasting memory?). Either way, this is fine to leave in this PR.

Comment (sharmarajdaksh) [reply bin/cli.js:158]: Hey, I was initially confused as well about what the reason to return the result was. In user usage, the returned result doesn't matter, but I think it gets used in certain tests so it should be kept in.

Comment (drwpow) [reply bin/cli.js:158]: OK. Yeah leave it in for now, and I can revisit that post-merge.

Comment (sharmarajdaksh) [reply bin/cli.js:158]: Alright! 
Rebasing and updating expected schemas fixed the broken test, so I think this PR should be ready now

**Comment by drwpow** [comment]
@all-contributors please add @sharmarajdaksh for code

**Pull request merged**